### PR TITLE
[EC-1062] Convert bar.js to TS and refactor

### PR DESCRIPTION
**Pull request by eliykat**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Getting in some low-hanging tech debt before starting on save prompt improvements. It'll be easiest to review this commit-by-commit to see each step individually. I also recommend hiding whitespace in the diff.

This is being merged into a feature branch to minimise QA's testing load, but it shouldn't be too long lived.

## Code changes

- convert `bar.js` to typescript (cherry-picked from @justindbaur, thank you!)
- use native [template elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) for the notification bar template content, instead of hiding it in hidden divs. Elements within templates are not returned by `getElementById` or `querySelector` calls (unless called explicitly on `template.content`), so we can use proper ids instead of misusing classnames to identify content. (and adding `-clone` to cloned content, etc etc)
- use [HTML data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) for the `isVaultLocked` attribute on the folder selector. This is what data attributes are intended for and is preferable to making up our own HTML attributes
- un-nest all the `bar.ts` functions. I assume these were nested in one big function to avoid namespace clashes, but [content scripts are already isolated](https://developer.chrome.com/docs/extensions/mv3/content_scripts/#isolated_world), so I don't think it's necessary.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: apps/browser/src/autofill/notification/bar.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+  // i18n for "Add" template
+  const addTemplate = document.getElementById("template-add") as HTMLTemplateElement;
+
+  const neverButton = addTemplate.content.getElementById("never-save");
+  neverButton.textContent = i18n.never;
+
+  const selectFolder = addTemplate.content.getElementById("select-folder");
+  selectFolder.setAttribute("aria-label", i18n.folder);
+  selectFolder.dataset.isVaultLocked = isVaultLocked.toString();
```
Comment (justindbaur) [inline apps/browser/src/autofill/notification/bar.ts:42]: TIL about how `.dataset.isVaultLocked` will automatically translate that to `data-is-vault-locked`.

File: apps/browser/src/autofill/notification/bar.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  const content = document.getElementById("content");
+  while (content.firstChild) {
+    content.removeChild(content.firstChild);
+  }
+
+  const newElement = template.content.cloneNode(true) as HTMLElement;
+  content.appendChild(newElement);
+}
+
+function sendPlatformMessage(msg: Record<string, any>) {
```
Comment (justindbaur) [inline apps/browser/src/autofill/notification/bar.ts:154]: nit: `Record<string, unknown>`

**Review by justindbaur — CHANGES_REQUESTED** [review]
Very nice, thanks for making this a reality. Love the `<template>` change and I just have one super tiny nit.

File: apps/browser/src/autofill/notification/bar.ts (hunk 2 above)
Comment (eliykat) [reply apps/browser/src/autofill/notification/bar.ts:154]: Good point, I don't use `unknown` a lot but I agree it's better practice here.

**Review by justindbaur — APPROVED** [review]

**Pull request merged**